sdks/java/fn-execution/src/test/java/org/apache/beam/sdk/fn/data/BeamFnDataGrpcMultiplexerTest.java

```diff
@@ -61,37 +63,64 @@
 
   @Test
   public void testOutboundObserver() {
-    Collection<BeamFnApi.Elements> values = new ArrayList<>();
-    BeamFnDataGrpcMultiplexer multiplexer = new BeamFnDataGrpcMultiplexer(
-        DESCRIPTOR,
-        (StreamObserver<BeamFnApi.Elements> inboundObserver)
-            -> TestStreams.withOnNext(values::add).build());
+    final Collection<BeamFnApi.Elements> values = new ArrayList<>();
+    BeamFnDataGrpcMultiplexer multiplexer =
+        new BeamFnDataGrpcMultiplexer(
+            DESCRIPTOR,
+            new StreamObserverClientFactory<Elements, Elements>() {
+              @Override
+              public StreamObserver<Elements> outboundObserverFor(
+                  StreamObserver<Elements> inboundObserver) {
+                return TestStreams.withOnNext(
```

Comment:
`values::add` instead of `Consumer<Elements>`?

Your choice if you still want to provide the StreamObserverClientFactory type instead of a lambda as visible in the old code.

Reply:
Done.